ext/nnstreamer/tensor_filter/tensor_filter_mediapipe.cc

```diff
@@ -0,0 +1,358 @@
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+};
+
+const char *mediapipe_subplugin::name = "mediapipe";
+const accl_hw mediapipe_subplugin::hw_list[] = {
+    ACCL_DEFAULT, ACCL_AUTO };
+
+/**
+ * @brief	mediapipe_subplugin Constructor
+ */
+mediapipe_subplugin::mediapipe_subplugin () :
+    tensor_filter_subplugin (),
+    config_path (nullptr)
+{
+  gst_tensors_info_init (&inputInfo);
+  gst_tensors_info_init (&outputInfo);
+
+  frame_timestamp = 0;
+}
+
+/**
+ * @brief	mediapipe_subplugin Destructor
+ */
+mediapipe_subplugin::~mediapipe_subplugin ()
+{
+  mediapipe::Status status;
+
+  g_free (config_path);
+
+  gst_tensors_info_free (&inputInfo);
+  gst_tensors_info_free (&outputInfo);
+
+  for (unsigned int i = 0; i < inputInfo.num_tensors; i++) {
+    status = graph.CloseInputStream (inputInfo.info[i].name);
+    if (!status.ok ()) {
+      std::cerr << "Failed to close input stream" << std::endl;
+    }
+  }
+  status = graph.WaitUntilDone ();
+  if (!status.ok ()) {
+    std::cerr << "Failed to closing mediapipe graph" << std::endl;
+  }
+}
+
+/**
+ * @brief	return empty instance
+ * @return an empty instance
+ */
+tensor_filter_subplugin & mediapipe_subplugin::getEmptyInstance ()
+{
+  return * (new mediapipe_subplugin ());
+}
+
+/**
+ * @brief	configure the instance before run it
+ * @param prop: the properties of the instance.
+ *              the properties are written with the construction of the pipeline
+ */
+void mediapipe_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  if (!prop->model_files[0] || prop->model_files[0][0] == '\0') {
+    std::cerr << "Model path is not given." << std::endl;
+    throw std::invalid_argument ("Model path is not given.");
+  }
+  assert (config_path == nullptr);
+  config_path = g_strdup (prop->model_files[0]);
+
+  gst_tensors_info_copy (&inputInfo, &prop->input_meta);
+  gst_tensors_info_copy (&outputInfo, &prop->output_meta);
+
+  if (loadMediapipeGraph (prop)) {
+    std::cerr << "Failed to load mediapipe graph" << std::endl;
+    throw std::runtime_error ("Failed to load mediapipe graph");
+  }
+}
+
+/**
+ * @brief	The mendatory delete function of mediapipe data pointer
+ */
+void mediapipe_subplugin::inputPtrDeleter (uint8_t* a) {
+  /* do nothing */
+  return;
+}
+
+/**
+ * @brief	load the mp model
+ * @note	the model will be loaded
+ * @return 0 if OK. non-zero if error.
+ *        -1 if the pbtxt file is not regular.
+ *        -2 if the pbtxt file is not loaded properly.
+ *        -3 if importing pbtxt file is failed at mediapipe.
+ *        -4 if init graph is failed.
+ */
+int
+mediapipe_subplugin::loadMediapipeGraph (const GstTensorFilterProperties * prop)
+{
+#if (DBG)
+  gint64 start_time = g_get_real_time ();
+#endif
+  gsize file_size;
+  gchar *content = nullptr;
+  GError *file_error = nullptr;
+  mediapipe::Status status;
+
+  g_assert (config_path != nullptr);
```

Comment:
unify assertion usage. use assert here or use g_assert in elsewhere.

Reply:
it's applied :)